ntp-daemon/src/server.rs

```diff
@@ -160,3 +154,86 @@ fn accept_packet(
         }
     }
 }
+
+#[derive(Debug)]
+struct TimestampedCache<T> {
+    elements: Vec<Option<(T, std::time::Instant)>>,
+}
+
+impl<T: std::hash::Hash + Eq> TimestampedCache<T> {
+    fn new(length: usize) -> Self {
+        Self {
+            // looks a bit odd, but prevents a `Clone` constraint
+            elements: std::iter::repeat_with(|| None).take(length).collect(),
```

Comment:
Try `vec![None; length]` or `[None; length].into_iter().collect()`?

Reply:
yeah, I thought that would work but it doesn't . `length` is not a constant so the array does not work. The vec just does not work. I tried adding a `    const EMPTY: Option<(T, Instant)> = None;` to the impl block but no luck.